### ISSUE TITLE
Is this correct behaviour when comparing versions

### ISSUE THREAD

**Issue by aslakagens** [vdurmont/semver4j#15]
Using     

`
compile 'com.vdurmont:semver4j:2.0.3'
`

`
        Semver A = new Semver("2.0", Semver.SemverType.LOOSE);
        Semver B = new Semver("2.0.0", Semver.SemverType.LOOSE);
        A.isGreaterThan(B);   // -> false
        B.isGreaterThan(A);   // -> false
        B.isEquivalentTo(A);  // -> false
        A.isEquivalentTo(B);  // -> false
        B.isLowerThan(A);     // -> true
        A.isLowerThan(B);     // -> true
`

**Comment by gihad** [comment]
Yes, I'm hit the same issue. The library doesn't know how to compare correctly when comparing a versions that don't have the same version type. e.g.:  "MAJOR" vs  "MAJOR.MINOR.PATCH"

**Comment by sschuberth** [comment]
I believe this should have been fixed by my PR https://github.com/vdurmont/semver4j/pull/20. Mind giving it a try?

**Comment by nomaed** [comment]
I still get 
```java
new Semver("2", SemverType.LOOSE).isEqualTo("2.0") // false
```

**Comment by sschuberth** [comment]
I actually had to do a fixup in PR https://github.com/vdurmont/semver4j/pull/25, which is not in a released version yet. *That* should fix it.

**Comment by nomaed** [comment]
I worked around it by adding a helper function that converts everything `.toStrict()`, so
```java
    private Semver version(String versionString) {
        return new Semver(versionString, SemverType.LOOSE).toStrict();
    }
```

But I don't like this workaround ;)

**Comment by collinpeters** [comment]
I assume this is another variant of this:
`new Semver("123", SemverType.NPM).satisfies("123") == false`

**Comment by mvdobrinin** [comment]
This is still an issue.


```
// should be true BUT IS FALSE
new Semver("2.0", Semver.SemverType.LOOSE).isGreaterThanOrEqualTo("2.0.0");

// should be true BUT IS FALSE
new Semver("2.0", Semver.SemverType.LOOSE).isEqualTo("2.0.0");

// should be true BUT IS FALSE
new Semver("2.0.0", Semver.SemverType.STRICT).isGreaterThanOrEqualTo(new Semver("2.0", Semver.SemverType.LOOSE));
```

**Comment by piotrooo** [comment]
@aslak-dirdal if you are still interesting, I've made [copy of this lib](https://github.com/semver4j/semver4j) and fix bug reported by you. Look for version 2.0.1
